### PR TITLE
feat(wasi) Implement WASI

### LINKED ISSUE #171: WASI in v1.0.0
### Summary

I can't see anything in the docs or source that would let me use WASI imports with the 1.0.0-x releases. Is this something that's going to be re-added? It's preventing me from upgrading to newer versions.

Cheers

### PR DESCRIPTION
**Pull request by Hywan**
Close #171
Close #72 
Close #128 
Depend on https://github.com/wasmerio/wasmer/pull/2053 (merged)
Depend on https://github.com/wasmerio/wasmer/pull/2058 (merged)
Depend on https://github.com/wasmerio/wasmer/pull/2059 (merged)
Depend on https://github.com/wasmerio/wasmer/pull/2070 (merged)

Work in Progress. It includes the `darwin/aarch64` branch. Will be rebased on `master` without that before merging of course.

### CODE REVIEW

File: wasmer/exports.go [hunk 1]
```diff
@@ -164,3 +167,13 @@ func (self *Exports) GetMemory(name string) (*Memory, error) {
 
 	return exports.IntoMemory(), nil
 }
+
+func (self *Exports) GetWasiStartFunction() (func(...interface{}) (interface{}, error), error) {
+	start := C.wasi_get_start_function(self.instance)
+
+	if start == nil {
+		return nil, newErrorWith("WASI start function is not found")
```
Comment (jubianchi) [inline wasmer/exports.go:175]:
```suggestion
		return nil, newErrorWith("WASI start function was not found")
```

**Review by jubianchi — APPROVED** [review]
seems good, only one minor suggestion.

**Comment by jubianchi** [comment]
bors try

**Comment by Hywan** [comment]
bors try

**Comment by Hywan** [comment]
bors try

**Pull request merged**